src/library-authoring/library-access/LibraryAccessForm.jsx

```diff
@@ -0,0 +1,180 @@
+                        value={data.email}
+                        onChange={onValueChange}
+                      />
+                    </ValidationFormGroup>
+                  </li>
+                </ol>
+              </fieldset>
+            </div>
+          </Card.Body>
+        </Card>
+        <Card className="mb-5">
+          <Card.Body>
+            <StatefulButton
+              variant="primary"
+              type="submit"
+              size="lg"
+              state={getSubmitButtonState()}
+              labels={{
+                disabled: intl.formatMessage(messages['library.access.form.button.submit']),
+                enabled: intl.formatMessage(messages['library.access.form.button.submit']),
+                pending: intl.formatMessage(messages['library.access.form.button.submitting']),
+              }}
+              icons={{
+                pending: <Icon className="fa fa-spinner fa-spin" />,
+              }}
+              disabledStates={['disabled', 'pending']}
+              className="action-button"
+            />
+            <Button size="lg" variant="secondary" className="mx-3 action-button" onClick={() => setShowAdd(false)}>Cancel</Button>
+          </Card.Body>
+        </Card>
+      </form>
+    </Row>
+  </>
+);
+
+LibraryAccessForm.propTypes = {
+  intl: intlShape.isRequired,
+  // eslint-disable-next-line react/forbid-prop-types
+  data: PropTypes.object.isRequired,
+  onValueChange: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  setShowAdd: PropTypes.func.isRequired,
+  hasFieldError: PropTypes.func.isRequired,
+  getFieldError: PropTypes.func.isRequired,
+  getSubmitButtonState: PropTypes.func.isRequired,
+};
+
+const initialFormState = () => ({
+  email: '',
+  access_level: LIBRARY_ACCESS.USER,
+});
+
+const LibraryAccessFormContainer = (
+  props,
+) => {
+  const [data, setData] = useState({
+    email: '',
+    access_level: LIBRARY_ACCESS.USER,
+  });
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    props.addUser({ libraryId: props.library.id, data }).then(() => {
+      setData(initialFormState());
+    });
+  };
+
+  const onValueChange = (event) => {
+    const { name, value } = event.target;
+    setData(current => (
+      {
+        ...current,
+        [name]: value,
+      }
+    ));
+  };
+
+  const hasFieldError = (fieldName) => !!(props.errorFields && (fieldName in props.errorFields));
+
+  const getFieldError = (fieldName) => {
+    if (hasFieldError(fieldName)) {
+      return props.errorFields[fieldName];
+    }
+    return null;
+  };
+
+  const getSubmitButtonState = () => {
+    let state;
+    if (props.submissionStatus === SUBMISSION_STATUS.SUBMITTING) {
+      state = 'pending';
+    } else if (data.email.match(/.+@.+[.].+/)) {
+      state = 'enabled';
+    } else {
+      state = 'disabled';
+    }
+    return state;
+  };
```

Comment:
This `getSubmitButtonState` does not need to be a function since it doesn't take any arguments. It should just be a string prop, and here you can just calculate its value inline in this `LibraryAccessFormContainer` function and pass the result as a string prop into `LibraryAccessForm`.

If the calculation was in any way expensive, you could wrap it in `useMemo` but it's just a trivial `if` statement so it's fine to run every time this `LibraryAccessFormContainer` function runs. That's better than what you have now, where every time this `LibraryAccessFormContainer` function runs, `getSubmitButtonState` becomes a new reference (to an identical function), causing `LibraryAccessForm` to render again because its `getSubmitButtonState` has technically changed to a new reference.